src/main/java/cloudy/Parser.java

```diff
@@ -0,0 +1,191 @@
+     * @param userInput The user's input as a string.
+     * @return A Command object representing the user's intended action.
+     */
+    public Command parseCommand(String userInput) {
+        if (userInput.startsWith("mark ")) {
+            return parseMarkCommand(userInput);
+        } else if (userInput.startsWith("unmark ")) {
+            return parseUnmarkCommand(userInput);
+        } else if (userInput.startsWith("todo")) {
+            return parseTodoCommand(userInput);
+        } else if (userInput.startsWith("deadline")) {
+            return parseDeadlineCommand(userInput);
+        } else if (userInput.startsWith("event")) {
+            return parseEventCommand(userInput);
+        } else if (userInput.startsWith("delete")) {
+            return parseDeleteCommand(userInput);
+        } else if (userInput.startsWith("find")) {
+            return parseFindCommand(userInput);
+        } else if (userInput.equals("list")) {
+            return new Command("list");
+        } else if (userInput.equals("bye")) {
+            return new Command("bye");
+        } else {
+            return new Command("invalid");
+        }
+    }
+
+    /**
+     * Parses a mark command from the user's input.
+     * @param userInput The user's input containing the mark command.
+     * @return A command object for marking a task as complete.
+     */
+    private Command parseMarkCommand(String userInput) {
+        String[] parts = userInput.split(" ");
+        if (parts.length >= 2) {
+            try {
+                int taskNumber = Integer.parseInt(parts[1]);
+                return new Command("mark", taskNumber);
+            } catch (NumberFormatException e) {
+                return new Command("invalidTaskNum");
+            }
+        } else {
+            return new Command("invalidTaskFormat");
+        }
+    }
+
+    /**
+     * Parses an unmark command from the user's input.
+     * @param userInput The user's input containing the mark command.
+     * @return A command object for marking a task as complete.
+     */
+    private Command parseUnmarkCommand(String userInput) {
+        String[] parts = userInput.split(" ");
+        if (parts.length == 2) {
+            try {
+                int taskNumber = Integer.parseInt(parts[1]);
+                return new Command("unmark", taskNumber);
+            } catch (NumberFormatException e) {
+                return new Command("invalidTaskNum");
+            }
+        } else {
+            return new Command("invalidCommand");
+        }
+    }
+
+    private Command parseTodoCommand(String userInput) {
+        if (userInput.trim().length() <= 4) {
+            return new Command("invalidCommand");
+        } else {
+            String taskDescription = userInput.substring(5).trim();
+            return new Command("todo", taskDescription);
+        }
+    }
+
+    private Command parseDeadlineCommand(String userInput) {
+        String[] parts = userInput.split("/by ");
```

Comment:
maybe `parts` can be replaced by a more meaningful name? For example `SplitedInputs`